core-bundle/src/Framework/ContaoFramework.php

```diff
@@ -114,6 +117,8 @@ public function initialize(bool $isFrontend = false): void
         if (!$this->legacyRouting) {
             $this->throwOnLegacyRoutingHooks();
         }
+
+        self::$nonce = random_bytes(6);
```

Comment:
```suggestion
        self::$nonce = bin2hex(random_bytes(16));
```

As some code might rely on valid UTF-8 code in templates we should only use “safe” ACII characters I think.

Reply:
Agreed. :+1: